drivers/adc/ad7124/ad7124.c

```diff
@@ -92,110 +85,98 @@ int32_t ad7124_no_check_read_register(struct ad7124_dev *dev,
 	 * in ADC_CONTROL, need to read 4, not 3 bytes for DATA with STATUS
 	 */
 	if ((p_reg->addr == AD7124_DATA_REG) &&
-	    (dev->regs[AD7124_ADC_Control].value & AD7124_ADC_CTRL_REG_DATA_STATUS)) {
+	    (dev->regs[AD7124_ADC_Control].value & AD7124_ADC_CTRL_REG_DATA_STATUS))
 		add_status_length = 1;
-	}
 
 	/* Read data from the device */
 	ret = no_os_spi_write_and_read(dev->spi_desc,
 				       buffer,
 				       ((dev->use_crc != AD7124_DISABLE_CRC) ? p_reg->size + 1
 					: p_reg->size) + 1 + add_status_length);
-	if(ret < 0)
+	if (ret)
 		return ret;
 
 	/* Check the CRC */
-	if(dev->use_crc == AD7124_USE_CRC) {
+	if (dev->use_crc == AD7124_USE_CRC) {
 		msg_buf[0] = AD7124_COMM_REG_WEN | AD7124_COMM_REG_RD |
 			     AD7124_COMM_REG_RA(p_reg->addr);
-		for(i = 1; i < p_reg->size + 2 + add_status_length; ++i) {
+		for (i = 1; i < p_reg->size + 2 + add_status_length; ++i)
 			msg_buf[i] = buffer[i];
-		}
 		check8 = ad7124_compute_crc8(msg_buf, p_reg->size + 2 + add_status_length);
 	}
 
-	if(check8 != 0) {
+	if (!check8)
```

Comment:
this looks like a functional change. Why `if (!check8)` instead of `if (check8)`

Reply:
This is one of the first comments I got in this PR as suggested by jsanbuen, I have changed it because if(check8 !=0)  is same as checking if (!check8).  Hope that makes sense.

Reply:
What I am trying to understand: which case should return an error code? `if (!check8)` or `if (check8)`? Because these 2 statements are completely different.

Reply:
It should actually be if(check8), will change it.